tools/gen-modules.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
+
+# Generate genhdr/modules.h for inclusion in py/objmodule.c.
+
+from __future__ import print_function
+
+import sys
+import os
+from glob import glob
+
+def update_modules(path):
+    modules = []
+    for module in sorted(os.listdir(path)):
+        if not os.path.isfile('%s/%s/module.mk' % (path, module)):
```

Comment:
So, this puts a restriction on the directory structure, in that your module name must match the directory it is in.  It's probably worth documenting this fact.

Reply:
> So, this puts a restriction on the directory structure, in that your module name must match the directory it is in. It's probably worth documenting this fact.

Good point, I've added it to the docs.